pkg/grpc/api_test.go

```diff
@@ -739,68 +768,274 @@ func TestGetSecret(t *testing.T) {
 				assert.Equal(t, resp.Data[tt.key], tt.expectedResponse, "Expected responses to be same")
 			} else {
 				assert.Error(t, err, "Expected error")
-				assert.Equal(t, codes.PermissionDenied, status.Code(err))
+				assert.Equal(t, tt.expectedError, status.Code(err))
 			}
 		})
 	}
 }
 
-func TestSaveState(t *testing.T) {
+func TestGetStateWhenStoreNotConfigured(t *testing.T) {
 	port, _ := freeport.GetFreePort()
+	server := startDaprAPIServer(port, &api{id: "fakeAPI"}, "")
+	defer server.Stop()
+
+	clientConn := createTestClient(port)
+	defer clientConn.Close()
 
-	server := startTestServer(port)
+	client := runtimev1pb.NewDaprClient(clientConn)
+	_, err := client.GetState(context.Background(), &runtimev1pb.GetStateRequest{})
+	assert.Equal(t, codes.FailedPrecondition, status.Code(err))
+}
+
+func TestSaveState(t *testing.T) {
+	fakeStore := &daprt.MockStateStore{}
+	fakeStore.On("BulkSet", mock.MatchedBy(func(reqs []state.SetRequest) bool {
+		if len(reqs) == 0 {
+			return false
+		}
+		return reqs[0].Key == "fakeAPI||key1"
```

Comment:
can we please change `key1`, `key2` to be something more descriptive so it's easier to connect when reading the test cases below?

Reply:
fixed.